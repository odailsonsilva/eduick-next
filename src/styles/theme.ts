export default {
  colors: {
    purple: {
      '100': '#CFC9E9',
      '500': '#6A40E4',
      '600': '#7A57FD',
      '900': '#200E64'
    },
    yellow: {
      '500': '#FFD74F '
    },
    white: {
      '500': '#ffffff',
      '900': '#E5E5E5'
    },
    black: {
      '500': '#343434'
    }
  },
  font: {
    family:
      "Poppins, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, sans-serif",
    regular: 'regular',
    bold: 'bold',
    sizes: {
      medium: '2.4rem',
      large: '5rem'
    }
  },
  spacings: {
    xxlg: '6rem',
    xlg: '4.8rem',
    lg: '3rem',
    md: '2rem',
    sm: '1.6rem',
    xsm: '0.5rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  }
}
