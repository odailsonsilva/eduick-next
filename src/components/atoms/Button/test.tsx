import { fireEvent, screen } from '@testing-library/react'
import { renderWithTheme } from '../../../utils/tests/helpers'
import theme from '../../../styles/theme'

import { Button } from '.'
import { transparentize } from 'polished'

describe('<Button />', () => {
  it('shold render button default', () => {
    renderWithTheme(<Button>Button</Button>)
    const button = screen.getByRole('button', { name: /Button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      background: theme.colors.yellow[500],
      color: '#6f48ec'
    })
  }),
    it('shold render button theme secondary', () => {
      renderWithTheme(<Button themeButton="secondary">Button</Button>)
      const button = screen.getByRole('button', { name: /Button/i })

      expect(button).toBeInTheDocument()
      expect(button).toHaveStyle({
        background: transparentize(0.7, theme.colors.purple[900]),
        color: theme.colors.white[500]
      })
    }),
    it('shold render button theme danger', () => {
      renderWithTheme(<Button themeButton="danger">Button</Button>)
      const button = screen.getByRole('button', { name: /Button/i })

      expect(button).toBeInTheDocument()
      expect(button).toHaveStyle({
        background: '#f44336',
        color: theme.colors.white[500]
      })
    }),
    it('shold render button theme danger', () => {
      renderWithTheme(<Button disabled>Button</Button>)
      const button = screen.getByRole('button', { name: /Button/i })

      expect(button).toBeInTheDocument()
      expect(button).toHaveStyle({ cursor: 'not-allowed' })
    }),
    it('shold render button clicks', () => {
      const onClick = jest.fn()

      const { container } = renderWithTheme(
        <Button onClick={onClick}>Button</Button>
      )
      const button = screen.getByRole('button', { name: /Button/i })

      fireEvent.click(button)

      expect(onClick).toHaveBeenCalled()

      expect(container.firstChild).toMatchSnapshot()
    })
})
