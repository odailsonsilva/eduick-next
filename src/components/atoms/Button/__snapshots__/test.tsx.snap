// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> shold render button clicks 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 4.8rem;
  font-size: 1.4rem;
  font-weight: bold;
  border: none;
  border-radius: 0.4rem;
  padding: 1.4rem 4.4rem;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  background: #FFD74F;
  color: #6f48ec;
}

.c0:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c0:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c0:active {
  -webkit-filter: brightness(1);
  filter: brightness(1);
}

<button
  class="c0"
>
  Button
</button>
`;
